app/validators/form_dob_validator.rb

```diff
@@ -0,0 +1,52 @@
+module FormDobValidator
+  extend ActiveSupport::Concern
+  include ActionView::Helpers::TranslationHelper
+
+  included do
+    include ActiveModel::Validations::Callbacks
+    validate :validate_dob
+  end
+
+  private
+
+  def validate_dob
+    return unless dob.present?
+    unless meet_min_age_req(dob)
+      errors.add(:dob, message: dob_min_age_error, type: :dob_min_age_error)
+    end
+  end
+
+  # @param [String|Date|ActionController::Parameters] dob: string or ActiveSupport::TimeWithZone
+  # @return [Boolean] false unless it meets the required minimal age,
+  #  including invalid input value and types
+  def meet_min_age_req(dob)
```

Comment:
Nit: since this returns only booleans, can you add '?' to the end of the method name?

Reply:
Aha, Ruby convention.

Reply:
Ideally the linter should catch it.